tests/fields/test_decimal.py

```diff
@@ -108,6 +155,37 @@ def test_decimal_places_bad(self):
             values[0], {"avg_decimal": Decimal("12.42")},
         )
 
+    async def test_aggregate_avg_with_f_expression(self):
+        await testmodels.DecimalFields.create(decimal=Decimal("0"), decimal_nodec=1)
+        await testmodels.DecimalFields.create(decimal=Decimal("9.99"), decimal_nodec=1)
+        await testmodels.DecimalFields.create(decimal=Decimal("27.27"), decimal_nodec=1)
+        values = (
+            await testmodels.DecimalFields.all()
+            .annotate(avg_decimal=Avg(F("decimal")))
+            .values("avg_decimal")
+        )
+        self.assertEqual(
+            values[0], {"avg_decimal": Decimal("12.42")},
+        )
+
+        values = (
+            await testmodels.DecimalFields.all()
+            .annotate(avg_decimal=Avg(F("decimal") + 1))
+            .values("avg_decimal")
+        )
+        self.assertEqual(
+            values[0], {"avg_decimal": Decimal("13.42")},
+        )
+
+        values = (
+            await testmodels.DecimalFields.all()
+            .annotate(avg_decimal=Avg(F("decimal") + F("decimal")))
```

Comment:
Could you do maths with one of the other fields? `decimal_nodec` or `decimal_null` instead. So we see that it works with both.

Reply:
Also, what is our expected behaviour when doing arithmetic on nulls? We probably want to ensure consistent behaviour at least?

Reply:
We can do with one of the other fields. i will add testcase to test that scenario.

Reply:
And doing arithmetic on nulls should generate null. But i'm not sure that all database(sqlite, postgres. etc) do equals.

Reply:
About nulls, I feel the most correct behaviour would to have the query fail? And then you'd have to explicitly set a defualt-if-null with `coalese()`?

At least it would be good to run the test with all db's to see what they do? And hope they do the same thing... At minumim just to document it?

Reply:
Could you see this [link](https://www.sqlite.org/nulls.html)? This links saids add anything to null is correct behavior. It should be okay just addding testcase with add something and null. 

Also i add test with add something with null. Does CI test it with all supported database? (e.g sqlite, postgres, etc)

Reply:
CI tests with all supported DB's yes. That's a handy link. It looks like the supported DB's should behave the same.